### PR TITLE
ToyCar - PBR Next Phase1 Sample Model

### PR DESCRIPTION
**Pull request by abwood**
This is the v4 sample model, with materials authored by @echadwick-wayfair.  This model is featured in the PBR Next Phase 1 Press Release, showcasing KHR_materials_clearcoat, KHR_materials_transmission, and KHR_materials_sheen.

### CODE REVIEW

**Comment by cx20** [comment]
Is this PR supposed to replace #268? If not, either or both of them are likely to have to change their names.

**Comment by abwood** [comment]
It is not intended to replace #268.  I would be more than happy to rename this model, but I couldn't come up with a name that made sense in this repository.  I had considered ToyCarPhase1, but I was concerned that the "Phase1" name only makes sense to the working group (these extensions represent phase 1 of PBR Next).  I'm open to any suggestions on a rename here.

If this gets merged before #268, then in my mind it would make sense to rename that model (e.g. ToyCarSpecular).

CC @proog128 or @emackey

**Comment by echadwick-wayfair** [comment]
How about ToyCarClearcoatSheenTransmission, or maybe with an underscore or
dash in there, e.g. ToyCar_ClearcoatSheenTransmission . Then for subsequent
revisions we could name them with whatever extensions are being demoed.

**Comment by emackey** [comment]
I would prefer not to rename from simply "ToyCar", although the model may need to move to the extensions section of the readme.

As for the other ToyCar in #268, those extensions are still under review & discussion.  I'm not sure that version of ToyCar demonstrates their effects in an intuitive way, like this version does with its extensions.  We might not even end up using a car to demonstrate those extensions when complete, depending how things go.

So my vote would be, this version of the ToyCar should get merged first (pending a bit more review), and samples featuring not-yet-ratified extensions can be worked out later on.  Sound good?

File: 2.0/ToyCar/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Toy Car
+
+## Screenshot
+
+![screenshot](screenshot/screenshot.jpg)
+
+## Description
+
+This model demonstrates the usage of `KHR_materials_sheen`, `KHR_materials_transmission` and `KHR_materials_clearcoat`.
```
Comment (bghgary) [inline 2.0/ToyCar/README.md:9]:
```suggestion
This model demonstrates the usage of [KHR_materials_sheen](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_materials_sheen/README.md), [KHR_materials_transmission](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_materials_transmission/README.md) and [KHR_materials_clearcoat](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_materials_clearcoat/README.md).
```

**Comment by bghgary** [comment]
Just a note: I tested this model in the Babylon.js sandbox with all of the extensions turned off and it still rendered properly. 👍  This model looks good to me barring some small comments.

image

**Review by bghgary — APPROVED** [review]

File: 2.0/ToyCar/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Toy Car
```
[the remark this answers is not in the document]
Comment (abwood) [reply 2.0/ToyCar/README.md:1]: We just caught that as well.  My mistake, I had a glb from old edits that snuck into my first commit.  This has been resolved.

File: 2.0/ToyCar/glTF/ToyCar.gltf [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+        "byteLength": 3664368
+    }],
+    "materials": [
+	{
+        "name": "ToyCar",
+        "doubleSided": true,
+        "pbrMetallicRoughness": {
+            "baseColorTexture": {
+                "index": 2
+			},
```
[the remark this answers is not in the document]
Comment (abwood) [reply 2.0/ToyCar/glTF/ToyCar.gltf]: Good catch.

**Review by emackey — APPROVED** [review]
I ran this ToyCar against the `dev` branch of the validator (2.0.0-dev.3.3 pre-release) which can validate the new PBR extensions, and it passed.  Also the model is awesome.  Merging.

**Comment by emackey** [comment]
> Ah, are you talking about the indents?

Alex fixed the indentation using a JSON pretty-formatter for VSCode.  This is merged now, thanks!

> Is Visual Studio Code the preferred editor for ascii glTF?

If you're asking me, the answer is so deep into "Yes" territory that I think it should probably be illegal to use other editing tools on raw glTF text.

**Comment by cx20** [comment]
@bghgary BTW, I tried to [insert](https://support.microsoft.com/en-us/office/get-creative-with-3d-models-ec5feb79-b0af-47f6-a885-151fcc88ac0a) this 3D model in `Excel` out of curiosity, and Excel crashed.
Probably because Excel does not support this glTF extension, but it should be modified so that it does not crash.

**Comment by bghgary** [comment]
Thanks @cx20! I've forwarded your message to the team who hopefully can help. It might be a while due to the holiday season.

**Pull request merged**